src/components/ActionBar/src/ActionBarButton.vue

```diff
@@ -49,27 +49,28 @@
 </template>
 
 <script>
-import chroma from 'chroma-js';
+import { colord } from 'colord';
 import PseudoWindow from 'vue-pseudo-window';
 import { MLoading } from '@square/maker/components/Loading';
 import { MThemeKey, defaultTheme, resolveThemeableProps } from '@square/maker/components/Theme';
 import { getContrast } from '@square/maker/utils/get-contrast';
 
-// TODO: refactor the code below so it's shared with Button component
+function setColorVariables(tokens) {
+	const textColor = getContrast(tokens.color, tokens.textColor);
```

Comment:
Minor nitpick - can we group the constants and the colors together in separate blocks to be more readable?

Reply:
Certainly, but what were you thinking specifically? I sort of approached it as:

1. base colors (color / background color)
2. state adjustments constants
3. state colors (with their conditionals)

I needed to declare the state adjustments before the state colors so they could be used there. Would you mind pseudo-code what you're thinking? I'd be happy to update in both components.

Reply:
Maybe just renaming the color constants to match their variable names? `const colorHover`, `const colorText` etc? Something scannable so it's more obvious at a glance which is a color generation constant and which are the colors that will be mapped to variables and returned.